File: 1-TwoSum/1-TwoSum.java
// Last updated: 4/24/2026, 8:15:32 AM
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; ++i) {
            int comp = target - nums[i];
            if (map.containsKey(comp)) {
                return new int[] {i, map.get(comp)};
            }
            map.put(nums[i], i);
        }
        return new int[] {};
    }
}

File: 1-TwoSum/1-TwoSum.cpp
// Last updated: 4/24/2026, 8:15:33 AM
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> mp;
        vector<int> res;
        for (int i = 0; i < nums.size(); ++i) {
            int complement = target - nums[i];
            if (mp.find(complement) != mp.end()) {
                res.push_back(i);
                res.push_back(mp[complement]);
                break;
            } else {
                mp[nums[i]] = i;
            }
        }
        return res;
    }
};